### ISSUE TITLE
Microsoft SP Site Request: Cannot parse the body of the sharepoint root site request

### ISSUE THREAD

**Issue by j574144** [gwsn/flysystem-sharepoint-adapter#11]
Hi.

Please help, I have received this error when try upload file

![image](https://github.com/gwsn/flysystem-sharepoint-adapter/assets/30190050/d5364fb5-6c93-466c-b975-5f08f426970a)

**Comment by jnovermars** [comment]
Can you try to isolate the upload of the file and make sure you have the correct Sharepoint Site and credentials?

```
<?php

use GWSN\FlysystemSharepoint\FlysystemSharepointAdapter;
use GWSN\FlysystemSharepoint\SharepointConnector;
use League\Flysystem\Filesystem;

$tenantId = '';
$clientId = '';
$clientSecret = '';
$sharepointSite = '';

$connector = new SharepointConnector($tenantId, $clientId, $clientSecret, $sharepointSite);

$prefix = '/test';
$adapter = new FlysystemSharepointAdapter($connector, $prefix);


$flysystem = new Filesystem($adapter);


$flysystem->write('test.txt', 'testContent');
```

**Comment by actengage** [comment]
I ran into this message when I was configuring this for my server. The credentialing for Sharepoint is quite painful. Check your SHAREPOINT_SITE and SHAREPOINT_PREFIX env vars, which is where I was going wrong. After following the setup instructions in the README to a "T" and it worked for me.

Edit, another trick I used was dd($accessToken) after line 26 in the GWSN\FlysystemSharepoint file to ensure you are getting an access token. If you get a token, you know your client key/secret is correct and move on to the site and prefix vars.

**Comment by jnovermars** [comment]
If you think you can improve the documentation to avoid those mistakes for the next users feel free to make a PR

Thanks for the feedback and glad the issue is resolved. 

Maybe adding a separate debug document with code samples will be a good addition. When I have time I will try to add this. 

And dd() will only work in combination with framework laravel or symfony (I know it requires the debug package).

**Comment by schaper1337** [comment]
I am getting a similar Error periodically, when trying to get a file from SharePoint within a queued job. 

"Microsoft SP Drive Request: Cannot parse the body of the sharepoint drive request. requestFileMetadata"

Everything is setup correctly, since downloading the same file from the frontend from sharepoint with this package works fine and usually the jobs also finisih without an exception... any clue what this could be or how to handle this?

**Comment by posidev-coding** [comment]
Ran in to this error briefly and it was a total goof.  In case this helps anyone else, make sure that your `SHAREPOINT_CLIENT_SECRET_VALUE` is actually using the `Value` for the Client Secret that you created.  I accidentally copied the `Secret ID`

**Comment by SpencerIsTheKey** [comment]
> I am getting a similar Error periodically, when trying to get a file from SharePoint within a queued job.
> 
> "Microsoft SP Drive Request: Cannot parse the body of the sharepoint drive request. requestFileMetadata"
> 
> Everything is setup correctly, since downloading the same file from the frontend from sharepoint with this package works fine and usually the jobs also finisih without an exception... any clue what this could be or how to handle this?

@schaper1337 Did you resolve this? I'm running into the same problem

**Comment by schaper1337** [comment]
no sorry! I reverted back to a local file driver for reliability reasons and didn't investigate the issue further...

Simon Bär
Geschäftsführer

Tel.: +49 40 80 80 5 88 11
Email: ***@***.***

K4S GmbH
Colonnaden 5  ·  20354 Hamburg

Amtsgericht: Hamburg  ·  HRB 175984
Geschäftsführer: Simon Bär

Datenschutz: https://k4s.de/about/privacy
Kontakt: https://k4s.de/about/contact

--

Diese E-Mail könnte vertrauliche und/oder rechtlich geschützte Informationen enthalten. 
Wenn Sie nicht der richtige Adressat sind oder diese E-Mail irrtümlich erhalten haben, 
informieren Sie bitte sofort den Absender und vernichten Sie diese Mail. 
Das unerlaubte Kopieren sowie die unbefugte Weitergabe dieser Mail sind nicht gestattet.

This e-mail may contain confidential and/or privileged information. 
If you are not the intended recipient (or have received this e-mail in error) please notify 
the sender immediately and destroy this e-mail. 
Any unauthorised copying, disclosure or distribution of the material in this e-mail is strictly forbidden.

________________________________
Von: Spencer Ke ***@***.***>
Gesendet: Thursday, May 8, 2025 10:04:53 PM
An: gwsn/flysystem-sharepoint-adapter ***@***.***>
Cc: Simon Bär ***@***.***>; Mention ***@***.***>
Betreff: Re: [gwsn/flysystem-sharepoint-adapter] Microsoft SP Site Request: Cannot parse the body of the sharepoint root site request (Issue #11)

[https://avatars.githubusercontent.com/u/38871528?s=20&v=4]SpencerIsTheKey left a comment (gwsn/flysystem-sharepoint-adapter#11)<https://github.com/gwsn/flysystem-sharepoint-adapter/issues/11#issuecomment-2864307933>

I am getting a similar Error periodically, when trying to get a file from SharePoint within a queued job.

"Microsoft SP Drive Request: Cannot parse the body of the sharepoint drive request. requestFileMetadata"

Everything is setup correctly, since downloading the same file from the frontend from sharepoint with this package works fine and usually the jobs also finisih without an exception... any clue what this could be or how to handle this?

@schaper1337<https://github.com/schaper1337> Did you resolve this? I'm running into the same problem

**Comment by alexthekiwi** [comment]
@SpencerIsTheKey any chance you happened to resolve this? I'm having the same issue. I'm also running writes to SharePoint from queued jobs, and this error pops up in my Sentry logs every now and then. It's intermittent too, all of my tests work – but then I'll just randomly get the odd one with this error.

**Comment by SpencerIsTheKey** [comment]
@alexthekiwi 

I did solve the issue, and it was a credentials problem. First it was incorrect credentials, then an issue with the Application's API permissions. To ensure that I was using the correct credentials, I tested the authorization request (POST to https://login.microsoftonline.com/{tenant-id}/oauth2/v2.0/token with client_id, scope, client_secret, and grant_type as body contents) with Postman to ensure everything was working properly. 

I'm using Laravel's queued jobs managed with Horizon, and I've also had the error come and go. It usually shows up around the time I have to renew the credentials or reissue the client secret. This is solved by restarting my Horizon queues, since the Laravel Workers are long-running processes that reflect the run environment at their creation time and don't update automatically. 

Hope this helps your issue
